### PR TITLE
Bucketed root repositories

### PR DESCRIPTION
**Pull request by jfontan**
Related to:

* https://github.com/src-d/borges/issues/180
* https://github.com/src-d/core-retrieval/pull/41

The command `packer` has bucketing disabled. I have not added a flag for this as `consumer` uses environment variables to change this parameter. Should this use environment variable or a flag for this specific command?

### CODE REVIEW

**Comment by erizocosmico** [comment]
Remember to update the core-retrieval dependency in `glide.yaml` and run `glide update` before merging this

File: archiver_test.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func (s *ArchiverSuite) SetupTest() {
 	s.tmpFs, err = fs.Chroot("tmp")
 	s.NoError(err)
 
-	s.tx = rrepository.NewSivaRootedTransactioner(rrepository.NewLocalCopier(s.rootedFs), s.txFs)
+	s.tx = rrepository.NewSivaRootedTransactioner(rrepository.NewLocalCopier(s.rootedFs, 0), s.txFs)
```
Comment (mcuadros) [inline archiver_test.go]: we don't have any test, with bucketing?

Comment (jfontan) [reply archiver_test.go]: It's tested in core-retrieval:

https://github.com/src-d/core-retrieval/blob/master/repository/repository_test.go#L71-L83

Comment (jfontan) [reply archiver_test.go]: I'll add a parameter for bucket size to the test and check both cases in borges.

**Comment by jfontan** [comment]
The changes are now rebased to master and tests both flat and bucketed hierarchy.

**Pull request merged**